components/org.wso2.carbon.identity.scim.provider/src/main/java/org/wso2/carbon/identity/scim/provider/impl/SCIMUserManager.java

```diff
@@ -396,7 +396,19 @@ public User getUser(String userId) throws CharonException {
                         continue;
                     }
 
-                    scimUser = this.getSCIMUserWithoutRoles(userName, new ArrayList<String>());
+                    List<String> claimURIList = new ArrayList<>();
+                    if (Boolean.parseBoolean(IdentityUtil.getProperty(SCIMProviderConstants
+                            .ELEMENT_NAME_SHOW_ALL_USER_DETAILS))) {
+                        // Get claims related to SCIM claim dialect.
+                        ClaimMapping[] claims = carbonClaimManager.
+                                getAllClaimMappings(SCIMCommonConstants.SCIM_CLAIM_DIALECT);
+                        for (ClaimMapping claim : claims) {
+                            claimURIList.add(claim.getClaim().getClaimUri());
+                        }
+                        scimUser = this.getSCIMUserWithoutRoles(userName, claimURIList);
+                    } else {
+                        scimUser = this.getSCIMUserWithoutRoles(userName, claimURIList);
```

Comment:
Do we really need if else condition?. Shall we change the logic to something like
if (Boolean.parseBoolean(IdentityUtil.getProperty(SCIMProviderConstants                     .ELEMENT_NAME_SHOW_ALL_USER_DETAILS))) {
assign values to claimURIList
}
scimUser = this.getSCIMUserWithoutRoles(userName, claimURIList);

So that we we can avoid to have else condition.